### PR TITLE
[TS] LPS-180849 Permission issue with shareable iframe fragments

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

Motivation
=======
The Guest user cannot see the contents of a shareable iframe, if it is put in a fragment.
The cause of the issue is that the Widget URL is pointing to the Draft Layout instead of the Published Layout.

Proposed Solution
========
Appending the Published Layout's friendly URL instead of the draft's URL, so that the shared iframe contains the right URL that can be seen by guest users too.

Steps to Verify
========
Please see the linked LPS for the reproduction steps.
https://issues.liferay.com/browse/LPS-180849

Thank you and best regards,
István

### CODE REVIEW

**Comment by catarineaguiar** [comment]
no regressions were found and I could not reproduce the bug with the fix.

File: modules/apps/portal/portal-util-test/src/testIntegration/java/com/liferay/portal/util/test/PortalImplWidgetURLResponseTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+		String portletId = LayoutTestUtil.addPortletToLayout(
+			draftLayout, JournalContentPortletKeys.JOURNAL_CONTENT);
+
+		ContentLayoutTestUtil.publishLayout(draftLayout, _layout);
+
+		Portlet portlet = PortletLocalServiceUtil.getPortletById(
+			_company.getCompanyId(), portletId);
+
+		URL widgetURL = new URL(_portal.getWidgetURL(portlet, themeDisplay));
```
Comment (tinatian) [inline modules/apps/portal/portal-util-test/src/testIntegration/java/com/liferay/portal/util/test/PortalImplWidgetURLResponseTest.java:93]: @IstvanD According to the steps in your ticket 

> Create a new public page and add the fragment to it, then publish it
> Log out and visit the page as guest user
You should not be able to see the same themeDisplay instance you used to publish layout, while in your test, you are doing `URL widgetURL = new URL(_portal.getWidgetURL(portlet, themeDisplay));` using the themeDisplay created with draftLayout.

This does not make sense to me.

Comment (IstvanD) [reply modules/apps/portal/portal-util-test/src/testIntegration/java/com/liferay/portal/util/test/PortalImplWidgetURLResponseTest.java:91]: @tinatian I pushed the changes.

**Comment by tinatian** [comment]
@IstvanD The test still does not make sense. In your test, you do:
1) Store the widgetURL
2) Publish Layout
3) Access the stored URL
This can not happen in real life according to your step "Log out and visit the page as guest user", as when you log out and visit, you should render the page again and have a new url generated based on the new themeDisplay, not the old URL you stored.

Please reconfirm if your issue is at PortalImpl#getServletURL(). If you can not create a reasonable test for it, most probably this is not root cause. 

Please resend.

Thanks.
Tina.

**Comment by IstvanD** [comment]
@tinatian in the scenario described by the reproduction steps, the URL generated by getWidgetURL() is stored in the fragment, and it is used when the Guest user renders the page.
What the test case does is exactly the same: generate and store the URL (in a variable) and then try to access it as the Guest user.

Please let me know if you need further explanation.

Thank you and best regards,
István

**Pull request closed without merging**